Source/Lib/Codec/EbEntropyCoding.c

```diff
@@ -5343,6 +5343,28 @@ static void CodeProfileTierLevel(
 	return;
 }
 
+/**************************************************
+* ComputeNumReorderPics
+**************************************************/
+EB_U32 ComputeNumReorderPics(SequenceControlSet_t    *scsPtr)
+{
+    // NumReorderPics indicates the maximum number of pictures that can precede any picture in decoding 
+    // order and follow it in output order.  The value needs to be no greater than the DPB size
+    //
+    // The reorderPicsPerHierarchy values are determined by inspection of the graphs at the top of 
+    // shown at the top of EbPredictionStructure.c and determining the highest observed value for 
+    // NumReorderPics in any one picture within that prediction structure.
+    //                    hierarchical level    1  2  3  4  5   6
+    static EB_U32 reorderPicsPerHierarchy[] = { 0, 1, 2, 4, 6, 14 };
```

Comment:
do you need it to be static? always visible? and it doesn't need to be U32.

Reply:
I guess your original thought is `const`?

Reply:
maxDpbSize and maxTemporalLayers are both U32 so might be best to keep the same as all three are used in the MIN macro

I can change to const and drop the static since it's inside a function.  Note: I don't think we're consistent on the use of constant throughout the code  (e.g. the prediction structures at the top of EbPredictionStructure are static (not const) ).

Reply:
We totally have 4-Level Hierarchies.
And level 4 has 3 NumReorderPics.
EB_U32 reorderPicsPerHierarchy[] = { 0, 1, 2, 3 };

Reply:
In level 4 hierarchical (referring to the graph in EbPredictionStructure.cpp), picture 5 has 4 pictures which precede it decoding order (3,2,4,1)  but follow it in output order.  That is the highest value of NumReorderPics I can see.

Reply:
We expose 4 hierarchical levels in the configuration interface, but the prediction structure code defines 6 levels (i.e. PredictionStructureConfigArray in EbPredictionStructure.cpp which is used to enumerate the number of levels in the PredictionStructureGroup constructor (PredictionStructureGroupCtor).

Reply:
Assume the number you are talking about is `display order` in line 175, pic 6 has pic 6,8 decode before it but display after it.

Reply:
On line 175 picture 1 in display order, needs pictures 2, 4, 6, and 8 decoded first, and these follow it in output order.  I think 4 makes sense for the MaxNumReorderPics value for this hierarchy.

Reply:
picture 1 doesn't need 6. pls check again. vicue is a better view.